Src/libCZI_UnitTests/test_azureblobstream.cpp

```diff
@@ -0,0 +1,187 @@
+    make_tuple(L"a=b x;c= d ;d=e  ;o=   w", map<wstring, wstring> { { L"a", L"b x" }, { L"c", L" d " }, { L"d", L"e  " }, { L"o", L"   w" } }),
+    make_tuple(L" a =b;c =d; d =e", map<wstring, wstring> { { L" a ", L"b" }, { L"c ", L"d" }, { L" d ", L"e" } }),
+    make_tuple(LR"(a=\;\;;c=\\d;d=e)", map<wstring, wstring> { { L"a", L";;" }, { L"c", LR"(\\d)" }, { L"d", L"e" } }),
+    make_tuple(LR"(c=\\d)", map<wstring, wstring> { { L"c", LR"(\\d)" } }),
+    make_tuple(LR"(\;a=abc)", map<wstring, wstring> { { LR"(;a)", L"abc" } }),
+    make_tuple(LR"(\;a\==abc)", map<wstring, wstring> { { LR"(;a=)", L"abc" } }),
+    make_tuple(LR"(c=\\d\=\;)", map<wstring, wstring> { { L"c", LR"(\\d=;)" } })
+));
+
+struct IllFormedAzureUriAndExpectedErrorFixture : public testing::TestWithParam<wstring> { };
+
+TEST_P(IllFormedAzureUriAndExpectedErrorFixture, TokenizeAzureUriScheme_InvalidCases)
+{
+    const auto parameter = GetParam();
+    EXPECT_THROW(Utilities::TokenizeAzureUriString(parameter), std::invalid_argument);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    AzureBlobStream,
+    IllFormedAzureUriAndExpectedErrorFixture,
+    testing::Values(
+    L"xxx",
+    L"=xxx",
+    LR"(a\=xxx)",
+    L";",
+    L"=",
+    L"=;",
+    LR"(\=\;)",
+    L"a=b;c=d;k="
+));
+
+static bool IsAzureBlobInputStreamAvailable()
+{
+    StreamsFactory::CreateStreamInfo create_info;
+    create_info.class_name = "azure_blob_inputstream";
+
+    for (int i = 0; i < StreamsFactory::GetStreamClassesCount(); ++i)
+    {
+        StreamsFactory::StreamClassInfo info;
+        if (StreamsFactory::GetStreamInfoForClass(i, info) && info.class_name == create_info.class_name)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static string EscapeForUri(const char* str)
+{
+    string result;
+    for (const char* p = str; *p; ++p)
+    {
+        if (*p == ';')
+        {
+            result += "\\;";
+        }
+        else if (*p == '=')
+        {
+            result += "\\=";
+        }
+        else
+        {
+            result += *p;
+        }
+    }
+
+    return result;
+}
+
+static const char* GetAzureBlobStoreConnectionString()
+{
+    // We use the environment variable 'AZURE_BLOB_STORE_CONNECTION_STRING' to communicate a connection string.
+
+    //const char* azure_blob_store_connection_string = std::getenv("AZURE_BLOB_STORE_CONNECTION_STRING");
+    //return azure_blob_store_connection_string;
+
+    
+    return R"(DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1;)";
+    
+}
+
+TEST(AzureBlobStream, GetStatisticsFromBlobUsingConnectionString)
+{
+    if (!IsAzureBlobInputStreamAvailable())
+    {
+        GTEST_SKIP() << "The stream-class 'azure_blob_inputstream' is not available/configured, skipping this test therefore.";
```

Comment:
```suggestion
        GTEST_SKIP() << "The stream-class 'azure_blob_inputstream' is not available/configured, therefore skipping this test.";
```

Sounds more idiomatic to me